Compiler warnings from -Wnarrowing

``` c++
engine/core/video/sdl/sdlimage.cpp: In member function ‘virtual void FIFE::SDLImage::useSharedImage(const ImagePtr&, const Rect&)’:
engine/core/video/sdl/sdlimage.cpp:682:63: warning: narrowing conversion of ‘region.FIFE::RectType<int>::x’ from ‘const int’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
   SDL_Rect srcrect = { region.x, region.y, region.w, region.h };
                                                               ^
engine/core/video/sdl/sdlimage.cpp:682:63: warning: narrowing conversion of ‘region.FIFE::RectType<int>::y’ from ‘const int’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
engine/core/video/sdl/sdlimage.cpp:682:63: warning: narrowing conversion of ‘region.FIFE::RectType<int>::w’ from ‘const int’ to ‘Uint16 {aka short unsigned int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
engine/core/video/sdl/sdlimage.cpp:682:63: warning: narrowing conversion of ‘region.FIFE::RectType<int>::h’ from ‘const int’ to ‘Uint16 {aka short unsigned int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
engine/core/video/image.cpp: In member function ‘virtual void FIFE::Image::copySubimage(uint32_t, uint32_t, const ImagePtr&)’:
engine/core/video/image.cpp:321:12: warning: narrowing conversion of ‘(((uint32_t)rect.FIFE::RectType<int>::x) + xoffset)’ from ‘uint32_t {aka unsigned int}’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
     rect.x + xoffset, rect.y + yoffset,
            ^
engine/core/video/image.cpp:321:30: warning: narrowing conversion of ‘(((uint32_t)rect.FIFE::RectType<int>::y) + yoffset)’ from ‘uint32_t {aka unsigned int}’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
     rect.x + xoffset, rect.y + yoffset,
                              ^
engine/core/video/image.cpp:326:57: warning: narrowing conversion of ‘rect.FIFE::RectType<int>::x’ from ‘const int’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
     SDL_Rect srcrect = { rect.x, rect.y, rect.w, rect.h };
                                                         ^
engine/core/video/image.cpp:326:57: warning: narrowing conversion of ‘rect.FIFE::RectType<int>::y’ from ‘const int’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
engine/core/video/image.cpp:326:57: warning: narrowing conversion of ‘rect.FIFE::RectType<int>::w’ from ‘const int’ to ‘Uint16 {aka short unsigned int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
a short unsigned int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
engine/core/video/image.cpp:334:46: warning: narrowing conversion of ‘xoffset’ from ‘uint32_t {aka unsigned int}’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
     static_cast<Uint16>(srcimg->getHeight()) };
                                              ^
engine/core/video/image.cpp:334:46: warning: narrowing conversion of ‘yoffset’ from ‘uint32_t {aka unsigned int}’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
engine/core/video/image.cpp:337:57: warning: narrowing conversion of ‘rect.FIFE::RectType<int>::x’ from ‘const int’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
     SDL_Rect srcrect = { rect.x, rect.y, rect.w, rect.h };
                                                         ^
engine/core/video/image.cpp:337:57: warning: narrowing conversion of ‘rect.FIFE::RectType<int>::y’ from ‘const int’ to ‘Sint16 {aka short int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
engine/core/video/image.cpp:337:57: warning: narrowing conversion of ‘rect.FIFE::RectType<int>::w’ from ‘const int’ to ‘Uint16 {aka short unsigned int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
engine/core/video/image.cpp:337:57: warning: narrowing conversion of ‘rect.FIFE::RectType<int>::h’ from ‘const int’ to ‘Uint16 {aka short unsigned int}’ inside { } is ill-formed in C++11 [-Wnarrowing]
```

--- a/engine/core/video/image.cpp
+++ b/engine/core/video/image.cpp
@@ -318,23 +318,34 @@ namespace FIFE {
 		if(this->isSharedImage()) {
 			Rect const& rect = this->getSubImageRect();
 			SDL_Rect dstrect = {
-				rect.x + xoffset, rect.y + yoffset,
+				static_cast<Sint16>(rect.x + xoffset),
+				static_cast<Sint16>(rect.y + yoffset),
 				static_cast<Uint16>(srcimg->getWidth()),
 				static_cast<Uint16>(srcimg->getHeight()) };
 			if(srcimg->isSharedImage()) {
 				Rect const& rect = srcimg->getSubImageRect();
-				SDL_Rect srcrect = { rect.x, rect.y, rect.w, rect.h };
+				SDL_Rect srcrect = {
+					static_cast<Sint16>(rect.x),
+					static_cast<Sint16>(rect.y),
+					static_cast<Uint16>(rect.w),
+					static_cast<Uint16>(rect.h) };
 				SDL_BlitSurface(srcimg->m_surface, &srcrect, m_surface, &dstrect);
 			} else {
 				SDL_BlitSurface(srcimg->m_surface, NULL, m_surface, &dstrect);
 			}
 		} else {
-			SDL_Rect dstrect = { xoffset, yoffset,
+			SDL_Rect dstrect = {
+				static_cast<Sint16>(xoffset),
+				static_cast<Sint16>(yoffset),
 				static_cast<Uint16>(srcimg->getWidth()),
 				static_cast<Uint16>(srcimg->getHeight()) };
 			if(srcimg->isSharedImage()) {
 				Rect const& rect = srcimg->getSubImageRect();
-				SDL_Rect srcrect = { rect.x, rect.y, rect.w, rect.h };
+				SDL_Rect srcrect = {
+					static_cast<Sint16>(rect.x),
+					static_cast<Sint16>(rect.y),
+					static_cast<Uint16>(rect.w),
+					static_cast<Uint16>(rect.h) };
 				SDL_BlitSurface(srcimg->m_surface, &srcrect, m_surface, &dstrect);
 			} else {
 				SDL_BlitSurface(srcimg->m_surface, NULL, m_surface, &dstrect);

--- a/engine/core/video/sdl/sdlimage.cpp
+++ b/engine/core/video/sdl/sdlimage.cpp
@@ -682,7 +682,11 @@ namespace FIFE {
 			src_surface->format->Bmask, src_surface->format->Amask);
 
 		SDL_SetAlpha(src_surface, 0, 0);
-		SDL_Rect srcrect = { region.x, region.y, region.w, region.h };
+		SDL_Rect srcrect = {
+			static_cast<Sint16>(region.x),
+			static_cast<Sint16>(region.y),
+			static_cast<Uint16>(region.w),
+			static_cast<Uint16>(region.h) };
 		SDL_BlitSurface(src_surface, &srcrect, surface, NULL);
 		SDL_SetAlpha(src_surface, SDL_SRCALPHA, 0);
 

